Source/Core/Core/AchievementManager.cpp

```diff
@@ -814,6 +817,15 @@ void AchievementManager::LoadGameCallback(int result, const char* error_message,
                                           rc_client_t* client, void* userdata)
 {
   AchievementManager::GetInstance().m_loading_volume.reset(nullptr);
+  if (result == RC_API_FAILURE)
```

Comment:
Is that really the best error code to detect this situation? Is there any situation where `RC_API_FAILURE` is emitted that doesn't mean 'your version is too old'?

Reply:
To my knowledge the only other things that would throw this error are a mismatch between what Dolphin is sending and what the server is expecting for requests, which would also mean the emulator is out of date. That said, this comment does make me consider that perhaps it's wiser to detect RC_NO_GAME_LOADED to send the game message and then have something more generic otherwise - but I was asked to make clear that the player's fix for this error is "update Dolphin".